code/DDSCodeTester.cpp

```diff
@@ -0,0 +1,1041 @@
+            return;
+        }
+
+        // Resetting the default PublisherQos to the original default constructed values
+        if(participant->set_default_publisher_qos(PUBLISHER_QOS_DEFAULT)
+                != ReturnCode_t::RETCODE_OK)
+        {
+            // Error
+            return;
+        }
+
+        // The previous instruction is equivalent to the following
+        if(participant->set_default_publisher_qos(PublisherQos())
+                != ReturnCode_t::RETCODE_OK)
+        {
+            // Error
+            return;
+        }
+        //!--
+    }
+}
+
+//DDS_DATAWRITER_LISTENER_SPECIALIZATION
+class CustomDataWriterListener : public DataWriterListener
+{
+
+public:
+
+    CustomDataWriterListener()
+    : DataWriterListener()
+    {
+    }
+
+    virtual ~CustomDataWriterListener()
+    {
+    }
+
+    virtual void on_publication_matched(
+            DataWriter* writer,
+            const PublicationMatchedStatus& info)
+    {
+        (void)writer
+        ;
+        if (info.current_count_change == 1)
+        {
+            std::cout << "Matched a remote Subscriber for one of our Topics" << std::endl;
+        }
+        else if (info.current_count_change == -1)
+        {
+            std::cout << "Unmatched a remote Subscriber" << std::endl;
+        }
+    }
+
+    virtual void on_offered_deadline_missed(
+             DataWriter* writer,
+             const OfferedDeadlineMissedStatus& status)
+    {
+         (void)writer, (void)status;
+         std::cout << "Some data could not be delivered on time" << std::endl;
+    }
+
+    virtual void on_offered_incompatible_qos(
+         DataWriter* writer,
+         const OfferedIncompatibleQosStatus& status)
+    {
+        (void)writer, (void)status;
+        std::cout << "Found a remote Topic with incompatible QoS" << std::endl;
+    }
+
+    /**
+    * @brief Method called when the livelivess of a publisher is lost
+    * @param writer The publisher
+    * @param status The liveliness lost status
+    */
+    virtual void on_liveliness_lost(
+         DataWriter* writer,
+         const LivelinessLostStatus& status)
+    {
+        (void)writer, (void)status;
+        std::cout << "Liveliness lost. Matched Subscribers will consider us offline" << std::endl;
+    }
+};
+//!--
+
+void dds_dataWriter_examples()
+{
+    // Taken out of the examples to avoid bloating them
+    DomainParticipant* participant =
+            DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+    Publisher* publisher =
+            participant->create_publisher(PUBLISHER_QOS_DEFAULT);
+    Topic* topic =
+            participant->create_topic("TopicName", "DataTypeName", TOPIC_QOS_DEFAULT);
+
+    {
+        //DDS_CREATE_DATAWRITER
+        // Create a DataWriter with default DataWriterQos and no Listener
+        // The symbol DATAWRITER_QOS_DEFAULT is used to denote the default QoS.
+        DataWriter* dataWriter_with_default_qos =
```

Comment:
I wouldn't mix camel and snake-case. Fast DDS style guide goes with snake, so I'd stick to that